src/Framework/EncodingUtilities.cs

```diff
@@ -274,10 +274,17 @@ internal static Encoding BatchFileEncoding(string contents, string encodingSpeci
             {
                 if (CurrentPlatformIsWindowsAndOfficiallySupportsUTF8Encoding())
                 {
-                    // Setting both encodings causes a change in the CHCP, making it so we don't need to P-Invoke CHCP ourselves.
-                    Console.OutputEncoding = Encoding.UTF8;
-                    // If the InputEncoding is not set, the encoding will work in CMD but not in PowerShell, as the raw CHCP page won't be changed.
-                    Console.InputEncoding = Encoding.UTF8;
+                    try
+                    {
+                        // Setting both encodings causes a change in the CHCP, making it so we don't need to P-Invoke CHCP ourselves.
+                        Console.OutputEncoding = Encoding.UTF8;
+                        // If the InputEncoding is not set, the encoding will work in CMD but not in PowerShell, as the raw CHCP page won't be changed.
+                        Console.InputEncoding = Encoding.UTF8;
+                    }
+                    catch (Exception ex) when (ex is IOException || ex is SecurityException)
```

Comment:
How did you arrive at these exceptions?

Reply:
The bug has the callstack that throws IOException. For SecurityException, I didn't get but referred to https://github.com/dotnet/msbuild/blob/5089df85a739c170a6c3ab0493bd2c17b36d81d4/src/MSBuild/AutomaticEncodingRestorer.cs#L62-L65. I think it's reasonable to not set encoding when no enough access.

Reply:
Looks good to me, though for the SecurityException it might be good to have low-pri message in binlog

Reply:
I don't think I'd do that now, though it would be kinda nice.